modules/controllers/jobs-controller.ts

```diff
@@ -31,27 +76,53 @@ export function JobsController(tracer?: any) {
             } = config[chain];
 
             // Guard against unconfigured chains
-            if (!balancerV3) {
+            if (!balancerV3 || !balancerPoolsV3 || !vaultAddress) {
                 throw new Error(`Chain not configured: ${chain}`);
             }
 
-            const vaultSubgraphClient = new BalancerVaultSubgraphSource(balancerV3);
-            const poolSubgraphClient = getPoolsSubgraphClient(balancerPoolsV3!);
+            const pools = await prisma.prismaPool.findMany();
+            const ids = pools.map((pool) => pool.id);
+            const client = getV3JoinedSubgraphClient(balancerV3, balancerPoolsV3);
+            const newPools = await client.getAllInitializedPools({ id_not_in: ids });
```

Comment:
this is clever!

Reply:
maybe safer to get all pools with paging and then filter here on node?

Reply:
getV3JoinedSubgraphClient – should handle paging – pools subgraph client needs to have the same paginated query as the vault has, i'll add it.